src/main/java/com/passpnu/passwordmanager/controller/ServiceController.java

```diff
@@ -1,4 +1,58 @@
 package com.passpnu.passwordmanager.controller;
 
+import com.passpnu.passwordmanager.dto.ServiceDto;
+import com.passpnu.passwordmanager.service.ServiceEntityService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.naming.NameNotFoundException;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/services")
+@AllArgsConstructor
 public class ServiceController {
+    private ServiceEntityService serviceEntityService;
+
+    @GetMapping
+    public List<ServiceDto> getServiceList(){
+        return serviceEntityService.getServiceList();
+    }
+
+    @PutMapping
+    public ResponseEntity<String> editService(@RequestBody ServiceDto service) throws NameNotFoundException {
+        if( !serviceEntityService.existsByDomain(service) ){
+            return new ResponseEntity<>("There are not such a service", HttpStatus.BAD_REQUEST);
+        }
+
+        Optional<ServiceDto> serviceDto = Optional.ofNullable(serviceEntityService.putService(service));
```

Comment:
Wrapping object in `Optional` just for null-check is a really expensive operation. A regular null-check will do the trick

Reply:
fixed